components/org.wso2.finance.open.banking.conformance.mgt/src/main/java/org/wso2/finance/open/banking/conformance/mgt/dao/impl/ReportDAOImpl.java

```diff
@@ -0,0 +1,277 @@
+            stmt.setInt(1, testID);
+            stmt.setString(2, userID);
+            stmt.setNull(3, Types.NULL);
+            stmt.setNull(4, Types.NULL);
+            stmt.executeUpdate();
+
+            ResultSet rs = stmt.getGeneratedKeys();
+            if(rs.next())
+            {
+                generatedReportID = rs.getInt(1);
+            }
+            stmt.close();
+            conn.close();
+        } catch(SQLException se) {
+            se.printStackTrace();
+        } catch(Exception e) {
+            e.printStackTrace();
+        } finally {
+            try{
+                if(stmt!=null) stmt.close();
+            } catch(SQLException se2) {
+            }
+            try {
+                if(conn!=null) conn.close();
+            } catch(SQLException se){
+                se.printStackTrace();
+            }
+        }
+        return generatedReportID;
+    }
+
+    /**
+     *This method will add a report to the row created in the report table
+     *at the start of a test.
+     * @param reportID : ID of the report
+     * @param report : Report object
+     */
+    @Override
+    public void updateReport(int reportID, Report report) {
+        Gson gson = new Gson();
+        Connection conn = DBConnector.getConnection();
+
+        java.util.Date dt = new java.util.Date();
+        java.text.SimpleDateFormat sdf = new java.text.SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        String currentTime = sdf.format(dt);
+
+        String reportJson = gson.toJson(report);
+        PreparedStatement stmt = null;
+        try {
+            String sql = SQLConstants.UPDATE_REPORT;
+            stmt = conn.prepareStatement(sql);
+            stmt.setString(1, reportJson);
+            stmt.setString(2, currentTime);
+            stmt.setInt(3, reportID);
+            stmt.executeUpdate();
+            stmt.close();
+            conn.close();
+        } catch(SQLException se) {
+            se.printStackTrace();
+        } catch(Exception e) {
+            e.printStackTrace();
+        } finally {
+            try{
+                if(stmt!=null) stmt.close();
+            } catch(SQLException se2) {
+            }
+            try {
+                if(conn!=null) conn.close();
+            } catch(SQLException se){
+                se.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     *This method will return a report object when the reportID is given.
+     * @param reportID : ID of the report
+     * @return the requested report
+     */
+    @Override
+    public Report getReport(int reportID) {
+        Gson gson = new Gson();
+        Report report = null;
+        Connection conn = DBConnector.getConnection();
+        PreparedStatement stmt = null;
+
+        try {
+            String sql = SQLConstants.RETRIEVE_REPORT;
+            stmt = conn.prepareStatement(sql);
+            stmt.setInt(1, reportID);
+            ResultSet rs = stmt.executeQuery();
```

Comment:
This resource is not properly closed on the checked exception. close it properly in the
catch(SQLException se) {
}